// See README.txt for information and build instructions.
//
// Note: START and END tags are used in comments to define sections used in
// tutorials.  They are not part of the syntax for Protocol Buffers.
//
// To get an in-depth walkthrough of this file and the related examples, see:
// https://developers.google.com/protocol-buffers/docs/tutorials

// [START declaration]
syntax = "proto3";
package tutorial;

import "google/protobuf/timestamp.proto";
// [END declaration]

// [START java_declaration]
//option java_package = "com.example.tutorial";
//option java_outer_classname = "AddressBookProtos";
// [END java_declaration]

// [START csharp_declaration]
//option csharp_namespace = "Google.Protobuf.Examples.AddressBook";
// [END csharp_declaration]

// [START messages]
message Student {
  string name = 1;
  int32 roll_no = 2;  // Unique ID number for this person.
 string email = 3;

  enum BookType {
    NOVEL = 0;
    FICTION = 1;
    NON_FICTION = 2;
  }

  message BookNumber {
    string id = 1;
    BookType type = 2;
    string bookname = 3;
  }

  repeated BookNumber books = 4;

  google.protobuf.Timestamp last_issue = 5;
}

// Our address book file is just one of these.
message LibraryRecords {
  repeated Student students = 1;
}
// [END messages]
